internal/manager/setup.go

```diff
@@ -210,15 +211,15 @@ func setupDataplaneAddressFinder(mgrc client.Client, c *Config, log logr.Logger)
 	return defaultAddressFinder, udpAddressFinder, nil
 }
 
-func buildDataplaneAddressFinder(mgrc client.Client, publishStatusAddress []string, publishServiceNn types.NamespacedName) (*dataplane.AddressFinder, error) {
+func buildDataplaneAddressFinder(mgrc client.Client, publishStatusAddress []string, publishServiceNn mo.Option[types.NamespacedName]) (*dataplane.AddressFinder, error) {
 	addressFinder := dataplane.NewAddressFinder()
 
 	if len(publishStatusAddress) > 0 {
 		addressFinder.SetOverrides(publishStatusAddress)
 		return addressFinder, nil
 	}
-	if publishServiceNn.String() != "" {
-		addressFinder.SetGetter(generateAddressFinderGetter(mgrc, publishServiceNn))
+	if serviceNn, ok := publishServiceNn.Get(); ok {
```

Comment:
I belive we've been consistent with using `NN` instead of `Nn`, across the codebase.